packages/react/src/components/List/ListHeader/ListHeader.jsx

```diff
@@ -35,6 +38,17 @@ const defaultProps = {
 };
 
 const ListHeader = ({ title, buttons, search, i18n, testId }) => {
+  const previousSearchValueRef = useRef(search?.value);
+  const hasFastSearch = search?.hasFastSearch !== false;
+  const handleSearch = (e) => {
+    const { value: currentValue } = e.target;
+    const { current: previousValue } = previousSearchValueRef;
+    if (previousValue !== currentValue) {
```

Comment:
Why is this needed? Is it to prevent multiple searches if the user hits enter multiple times, or is it a fix for the `handleSearch` being called multiple times due to some other reason?

Reply:
Exactly. If you hit enter, but then blur the input the handler is called twice. This prevents that if nothing has changed.